File: 프로그래머스/2/155651. 호텔 대실/호텔 대실.py
def solution(book_time):
    answer = 0
    booklist = []
    for i in range(len(book_time)) :
        book_time[i][0]=book_time[i][0][0]+book_time[i][0][1]+book_time[i][0][3]+book_time[i][0][4]
        book_time[i][1]=book_time[i][1][0]+book_time[i][1][1]+book_time[i][1][3]+book_time[i][1][4]

        #:제거했음
    
    book_time.sort(key= lambda x:x[:][0])
    #퇴실이 짧은애부터 정렬
    
    
    print(book_time)
    
    roomlist = []
    for i in range(len(book_time)) :
        if i == 0 :
            roomlist.append(int(book_time[i][1]))
            continue
        else:
            isthere = False
            for j in range(len(roomlist)) :
                nextroom = roomlist[j]+10 
                if nextroom%100 >=60 :
                    nextroom = (nextroom//100+1)*100 + (nextroom%100-60)
                if int(book_time[i][0]) >= nextroom :
                    roomlist[j] = int(book_time[i][1])
                    isthere = True
                    break
            
            if isthere== False:
                roomlist.append(int(book_time[i][1]))
            roomlist.sort()    
                
        
    
    return len(roomlist)

File: 프로그래머스/2/155651. 호텔 대실/호텔 대실.java
import java.util.*;

class Solution {
    public int solution(String[][] book_time) {
        int answer = 0;
        PriorityQueue<int []> pq = new PriorityQueue<>(
        (a,b) -> 
            {
                return a[0]-b[0];
            });
        
        for (int i = 0 ; i<book_time.length; i++)
        {
            String now = book_time[i][0];
            String [] firstarr = now.split(":");
            int totaltime1 = 0;
            totaltime1+= Integer.parseInt(firstarr[0])*60;
            totaltime1+= Integer.parseInt(firstarr[1]);
            
            String [] secondarr = book_time[i][1].split(":");
            int totaltime2 = 0;
            totaltime2+= Integer.parseInt(secondarr[0])*60;
            totaltime2+= Integer.parseInt(secondarr[1]);
            
            pq.add(new int [] {totaltime1,totaltime2});
        }
        
        PriorityQueue<int []> pq2 = new PriorityQueue<>(
        (a,b) -> 
            {
                return a[1]-b[1];
            });
        
        
        while(!pq.isEmpty())
        {
            int [] now = pq.poll();
            
            while(!pq2.isEmpty() && pq2.peek()[1] <= now[0] )
            {
                pq2.poll();
            }
            
            int [] newarr = new int [] {now[0],now[1]+10};
            
            pq2.add(newarr);
            answer = Math.max(pq2.size(),answer);
            
        }
        
        return answer;
    }
}